system-contracts/contracts/PubdataChunkPublisher.sol

```diff
@@ -24,27 +25,21 @@ contract PubdataChunkPublisher is IPubdataChunkPublisher {
 
         blobLinearHashes = new bytes32[](blobCount);
 
-        // We allocate to the full size of blobCount * BLOB_SIZE_BYTES because we need to pad
-        // the data on the right with 0s if it doesn't take up the full blob
-        bytes memory totalBlobs = new bytes(BLOB_SIZE_BYTES * blobCount);
-
-        assembly {
-            // The pointer to the allocated memory above. We skip 32 bytes to avoid overwriting the length.
-            let ptr := add(totalBlobs, 0x20)
-            calldatacopy(ptr, _pubdata.offset, _pubdata.length)
-        }
-
+        uint256 ptr;
         for (uint256 i = 0; i < blobCount; ++i) {
-            uint256 start = BLOB_SIZE_BYTES * i;
-
-            bytes32 blobHash;
-            assembly {
-                // The pointer to the allocated memory above skipping the length.
-                let ptr := add(totalBlobs, 0x20)
-                blobHash := keccak256(add(ptr, start), BLOB_SIZE_BYTES)
+            if (ptr + BLOB_SIZE_BYTES <= _pubdata.length) {
+                // Pass chunk by pointer since we don't need to pad it
+                blobLinearHashes[i] = EfficientCall.keccak(_pubdata[ptr:ptr + BLOB_SIZE_BYTES]);
+            } else {
+                // Copy to memory to pad with zeroes
+                bytes memory blob = new bytes(BLOB_SIZE_BYTES);
+                assembly {
+                    calldatacopy(add(0x20, blob), add(_pubdata.offset, ptr), BLOB_SIZE_BYTES)
```

Comment:
here you copy `BLOB_SIZE_BYTES`, it assumes that after `_pubdata` there is nothing in calldata and so zeroes will be used here. This is true today, but in the future if we'll ever change the params, it will break and be very hard to debug, please copy only the needed bytes

Reply:
Done